agent/src/agent/pipeline/config/jython_scripts/promql.py

```diff
@@ -75,44 +76,56 @@ def get_metric_name(record):
 
 def process_matrix(result_, end_):
     batch = sdc.createBatch()
-    for res in result_['data']['result']:
-        base_record = dict(res['metric'].items())
-        for timestamp, value in res[get_result_key(result_)]:
-            record = base_record.copy()
-            record['timestamp'] = float(timestamp)
+    if result_['data']['result']:
```

Comment:
I guess you can move this check to the main function because they're the same here and in the `process_vector`